models/models/markov_chain/markov_chain.py

```diff
@@ -34,28 +37,31 @@ def train(self, epochs: int, x_train: Any, y_train: Any, progress_callback: Prog
         for i in range(n_gram_next.shape[0]):
             n_gram_next[i] = []
 
+        start = time.time()
+        time.perf_counter()
+
         for i in range(len(self.data)):
+
+            elapsed = time.time() - start
+            progress_callback([(elapsed, 100 * i / len(self.data))])
```

Comment:
How large can `len(self.data)` get? I added the if-statement over the progress callback because it used to spam a LOT of checkpoints (in thousands).

Reply:
In Bach dataset it's about 600